state/state.go

```diff
@@ -104,7 +107,7 @@ type State struct {
 	common.Name
 	StateConfig
 	logging           *logging.LayerLogger
-	Pub               // Publisher hooks
+	Pub               *pubregistry.PubRegistry // Publisher hooks for this vm
```

Comment:
I moved "Pub" out of state to a separate pubregistry package/struct because the growing number of publishers in the registry started to pollute the already severely polluted state package. (And vice versa)
Placing it in the pubsub package will cause import cycle errors, plus a surplus is that in IntelliJ/golang the drop-down list for state.Pub. is cvery clean.

Reply:
totally agree w/ this motivation here

a of the choices we're making with the WAX rewrite are intended to ease the import cycle burden
while also making life nicer for us in the IDE